facade/nodeFacade.go

```diff
@@ -198,27 +211,17 @@ func (nf *nodeFacade) startRest() {
 	}
 }
 
-func (nf *nodeFacade) createMiddlewareLimiters() ([]api.MiddlewareProcessor, error) {
-	sourceLimiter, err := middleware.NewSourceThrottler(nf.wsAntifloodConfig.SameSourceRequests)
-	if err != nil {
-		return nil, err
-	}
-	go nf.sourceLimiterReset(sourceLimiter)
-
-	globalLimiter, err := middleware.NewGlobalThrottler(nf.wsAntifloodConfig.SimultaneousRequests)
-	if err != nil {
-		return nil, err
-	}
-
-	return []api.MiddlewareProcessor{sourceLimiter, globalLimiter}, nil
-}
-
 func (nf *nodeFacade) sourceLimiterReset(reset resetHandler) {
+	betweenResetDuration := time.Second * time.Duration(nf.wsAntifloodConfig.SameSourceResetIntervalInSec)
 	for {
-		time.Sleep(time.Second * time.Duration(nf.wsAntifloodConfig.SameSourceResetIntervalInSec))
-
-		log.Trace("calling reset on WS source limiter")
-		reset.Reset()
+		select {
+		case <-time.After(betweenResetDuration):
```

Comment:
So this is a "hard" reset, which means from time to time we forgive everyone, which means that an eventual attacker, if starting right before the reset and continuing immediately after a reset, could send `maxNumRequestsPerAddress * 2` requests in a short period of time (just thinking out loud, shouldn't be an issue, sliding mechanism not really required).